lms/static/js/spec/dateutil_factory_spec.js

```diff
@@ -0,0 +1,50 @@
+define(['../dateutil_factory.js'], function(DateUtilIterator) {
+    'use strict';
+
+    describe('DateUtilFactory', function() {
+        beforeEach(function() {
+            setFixtures('<div class="test"></div>');
+        });
+
+        describe('stringHandler', function() {
+            it('returns a complete string', function() {
+                var localTimeString = 'RANDOM_STRING';
+                var sidecarString = 'RANDOM_STRING_TWO {random_token}';
+                var dateToken = 'random_token';
+                var answer = 'RANDOM_STRING_TWO RANDOM_STRING';
+                expect(DateUtilIterator.stringHandler(localTimeString, sidecarString, dateToken)).toEqual(answer);
+            });
+        });
+
+        describe('transform', function() {
+            var $form;
+
+            it('localizes some times', function() {
+                /* we have to generate a fake span and then test the resultant texts */
+                var iterationKey = '.localized-datetime';
+                var TestLangs = {
```

Comment:
Nit: the variable name should be in camelCase, per convention.  I'm surprise you didn't get an eslint warning about this.